feat: Introduce `jobKind` property for job type distinction

ce-dmelnych opened the pull request:
## Description:
This PR introduces a new `jobKind` property to the `JobRecordValue` and `JobRecord` classes within the Zeebe workflow engine. The addition of the `jobKind` property, an Enum comprising `BPMN_ELEMENT`, `EXECUTION_LISTENER`, and `TASK_LISTENER`, enables the Zeebe engine to distinguish between different types of jobs.

### Key Changes:
- Added the `jobKind` property to `JobRecordValue` and `JobRecord` as an Enum.
- Updated the `BpmnJobBehavior` class to set `jobKind` to BPMN_ELEMENT by default for all jobs, ensuring backward compatibility.
- Modified Elasticsearch/OpenSearch exporter configuration files to include the new `jobKind` property.

## Related issues

closes #16208

<!-- Cut-off marker
_All lines under and including the cut-off marker will be removed from the merge commit message_

## Definition of Ready

Please check the items that apply, before requesting a review.

You can find more details about these items in our wiki page about [Pull Requests and Code Reviews](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews).

* [ ] I've reviewed my own code
* [ ] I've written a clear changelist description
* [ ] I've narrowly scoped my changes
* [ ] I've separated structural from behavioural changes
-->

## Definition of Done

<!-- Please check the items that apply, before merging or (if possible) before requesting a review. -->

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

saig0 reviewed (approved):
@ce-dmelnych well done. :medal_sports: 

:wrench: I miss only one minor thing: adjusting the `CompactRecordLogger` ([ref](https://github.com/camunda/zeebe/blob/main/docs/developer_handbook.md#how-to-extend-an-existing-record)). The logger prints a record summary, for example, for the test output. It would be nice to extend the [summary for jobs](https://github.com/camunda/zeebe/blob/main/test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java#L414) by the job kind.

ce-dmelnych commented:
Thank you for the feedback and the recommendation, @saig0! 👍🙏

I've incorporated your suggestion and updated the `CompactRecordLogger` to include the job kind in its summary. 
Now it prints job details in the following way:
```
E JOB CREATED - #18->#04 K08 - K08 "test" @"task"[K07] (EXECUTION_LISTENER), 5 retries, in <process "process"[K03]> (no vars)
```

Please let me know if you'd prefer to have another formatting, and I'll be happy to make further adjustments.

saig0 commented:
@ce-dmelnych looks good. :rocket:
